controller/workitem_blackbox_test.go

```diff
@@ -969,6 +969,114 @@ func (s *WorkItem2Suite) TestWI2UpdateSetReadOnlyFields() {
 	})
 }
 
+func (s *WorkItem2Suite) TestWI2UpdateWorkItemType() {
+	fxt := tf.NewTestFixture(s.T(), s.DB,
+		tf.CreateWorkItemEnvironment(),
+		tf.Users(2),
+		tf.Identities(2, func(fxt *tf.TestFixture, idx int) error {
+			fxt.Identities[idx].User = *fxt.Users[idx]
+			return nil
+		}),
+		tf.WorkItemTypes(2, func(fxt *tf.TestFixture, idx int) error {
+			switch idx {
+			case 0:
+				fxt.WorkItemTypes[idx].Fields = map[string]workitem.FieldDefinition{
+					"fooo": {
```

Comment:
That won't work. The make function creates a new space template and then new workitem type. The new workitem type doesn't have any fields of planner type. 
Instead if I use agile/sdd template while creating the space, the workitem types would be available and then I can test actual template types.
I'll add new tests for agile and sdd space template.

Reply:
@kwk Do we really need tests for agile/sdd template (aka Planner Types)? The agile workitem types aren't directly available. I'll have to load them from the DB and then change the type. This might complicate the actual test.

Reply:
@jarifibrahim all WITs created by the test fixture DO have all fields as defined by the planner item base type. The following line causes the WITs to extend that type:

https://github.com/fabric8-services/fabric8-wit/blob/19c0e19355ac3b0aa8bdd67e8a9e0854a3843ed2/test/testfixture/make_functions.go#L302

Reply:
Based on the discussion with @kwk yesterday, @kwk will add tests for this.